lib/thrift/protocols/binary.ex

```diff
@@ -99,6 +116,22 @@ defmodule Thrift.Protocols.Binary do
     <<1::size(1), 1::size(15), 0::size(8),
     # ^^ Strange, I know. We could integrate the 8-bit zero here with the 5 bit zero below.
     0::size(5), to_message_type(message_type)::size(3),
-    byte_size(name)::32-signed, sequence_id::32-signed>>
+    byte_size(name)::32-signed, name::binary, sequence_id::32-signed>>
+  end
+
+  def deserialize(:message_begin,<<1::size(1), 1::size(15), 0::size(8),
```

Comment:
That third byte is undefined according to the "spec" we're working from. I'd just leave it an unmatched `_` instead of requiring it to be zero.